components/generate_and_detect_component.py

```
@@ -0,0 +1,11 @@
+import streamlit as st
+from utils.extractor import extract_article_text
+from utils.language import detect_language
+
+def generate_and_detect():
+    with st.spinner("추출 중입니다..."):
+        text = extract_article_text(url)
```

Ruff (0.11.9)

Undefined name `url`

(F821)